daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SExpr.scala

```diff
@@ -102,7 +102,13 @@ object SExpr {
 
   object SEValue extends SValueContainer[SEValue] // used by Compiler
 
-  final case class SEAppGeneral(fun: SExpr, args: Array[SExpr]) extends SExpr with SomeArrayEquals {
+  /** Function application with general function/arguments (deprecated)
+    * This case exists purely for use by:
+    * and that us in turn is to support the current stack-trace support, which peeks under KArg.
+    */
+  final case class SEAppGeneral_DEPRECATED(fun: SExpr, args: Array[SExpr])
```

Comment:
```suggestion
    @deprecated("Should not be used outside blabla....")
    final case class SEAppGeneral(fun: SExpr, args: Array[SExpr])
```
together with the annotation 
```
@nowarn("""cat=deprecation&origin=com.daml.lf.speedy.SExpr.SEAppOnlyFunIsAtomic""")
```
the line before any function that use it.